recipes/stx/all/conanfile.py

```diff
@@ -0,0 +1,147 @@
+    settings = 'os', 'compiler', 'build_type', 'arch'
+    options = {
+        'backtrace': [True, False],
+        'panic_backtrace': [True, False],
+        'override_panic_handler': [True, False],
+        'shared': [True, False],
+        'fPIC': [True, False],
+        'visible_panic_hook': [True, False],
+    }
+    default_options = {
+        'backtrace': False,
+        'override_panic_handler': False,
+        'shared': False,
+        'fPIC': True,
+    }
+    exports_sources = ['CMakeLists.txt', 'patches/*']
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.panic_backtrace == None:
+            self.options.panic_backtrace = (
+                self.options.backtrace and
+                not self.options.override_panic_handler
+            )
+
+        if self.options.visible_panic_hook == None:
+            self.options.visible_panic_hook = self.options.shared
+
+        compiler = self.settings.compiler
+        compiler_version = tools.Version(self.settings.compiler.version)
+        standards = ['17', '20', 'gnu17', 'gnu20']
+
+        if compiler.cppstd and not str(compiler.cppstd) in standards:
+            raise ConanInvalidConfiguration('STX requires C++17 support')
+
+        if compiler == 'Visual Studio' and compiler_version < 16:
+            raise ConanInvalidConfiguration(
+                'STX requires C++17 support, use at least VS 2019'
+            )
+
+        if compiler == 'gcc' and compiler_version < 8:
+            raise ConanInvalidConfiguration(
+                'STX requires C++17 support, use at least GCC 8'
+            )
+
+        if (compiler == 'clang' and compiler.libcxx and
+                compiler.libcxx in ['libstdc++', 'libstdc++11'] and
+                compiler_version < 9):
+            raise ConanInvalidConfiguration(
+                'STX requires C++17 language and standard library features '
+                'which clang & libc++ < 10 lack'
+            )
+
+        if (compiler == 'clang' and compiler.libcxx and
+                compiler.libcxx == 'libc++' and
+                compiler_version < 10):
+            raise ConanInvalidConfiguration(
+                'STX requires C++17 language and standard library features '
+                'which clang & libc++ < 10 lack'
+            )
+
+        if compiler == 'apple-clang':
+            raise ConanInvalidConfiguration(
+                'STX requires C++17 language and standard library features '
+                'which apple-clang and libc++ lack'
+            )
+
+        if (compiler == 'Visual Studio' and self.options.shared and
+                tools.Version(self.version) <= '1.0.1'):
+            raise ConanInvalidConfiguration(
+                'shared library build does not work on windows with '
+                'STX version <= 1.0.1'
+            )
```

Comment:
I would suggestion removing 1.0.0 since there little reason to use it

Reply:
Will do, but this <= 1.0.1 check is still necessary as the fix for this configuration not building is not yet released.

Reply:
Ahh! I miss-read the `<=` as `<` you are right